compiler/semobjconstr.nim

```diff
@@ -84,6 +84,123 @@ proc caseBranchMatchesExpr(branch, matched: PNode): bool =
 
   return false
 
+type
+  TRanges = seq[Slice[BiggestInt]]
```

Comment:
Seem overkill, we have ``IntSet`` which while lacking the support for the ``Slice`` aspect is much easier to deal with. Keep in mind that object discriminants are almost always enums or bools with possible values < 256. IntSets can handle these just fine.

Reply:
alright, done.